Update deprecation policy (#10839)
flutter/website: src/content/release/compatibility-policy.md

which is exclusively based on whether submitted tests fail, as
described above.

> Deprecated APIs are not removed on a scheduled basis. If any are removed,
> we follow the same procedures listed above for making breaking changes
> in removing the deprecated API.

Reviewer: **issue:** Cleaned up based on Google Style rules. Write in active voice per [GSG](https://developers.google.com/style/voice). (Also, some word list issues.)
Also, the 

```suggestion
The Flutter team doesn't remove deprecated APIs on a scheduled basis.
If the team removes a deprecated API,
it follows the same procedures as those for breaking changes.
```
